### PR TITLE
Object cache just the post's translations IDs rather than objects

### PR DESCRIPTION
**Pull request by joehoyle**
This means we don't have to use up space unnecessarily, and also in some
case this will mean there is no DB activity at all due to a persistant object
cache and no need for additional WP_Query'ies

See #236

### CODE REVIEW

File: class-post-public.php [hunk 1]
```diff
@@ -1221,20 +1223,11 @@ public function get_post_translations( $post ) {
...
 
-		// Get all the translations in one cached DB query
-		$args = array(
-			// We want a clean listing, without any particular language
-			'bbl_translate' => false,
-			'include' => $post_ids,
-			'post_type' => $post_types,
-			'post_status' => array( 'publish', 'pending', 'draft', 'future' ),
-		);
-		$posts = get_posts( $args );
```
Comment (simonwheatley) [inline class-post-public.php:1232]: Is there advantage in still getting all posts in the translation group, as IDs only in line with the other changes in this PR, in one DB operation?

**Comment by simonwheatley** [comment]
Are there some quick basic unit tests which would be appropriate here?

**Comment by simonwheatley** [comment]
> Are there some quick basic unit tests which would be appropriate here?

In which case, we should wait until #271 is merged.

**Comment by joehoyle** [comment]
- [x] Add unit tests

**Comment by simonwheatley** [comment]
@joehoyle Any update here?

**Comment by johnbillion** [comment]
Over to you, Simon. Fixes and tests added to the PR.

**Pull request merged**